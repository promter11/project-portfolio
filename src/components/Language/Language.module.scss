@import "../../styles/_variables";
@import "../../styles/_mixins";

.language {
  position: relative;

  .languageEnter,
  .languageExit {
    opacity: 0;
    transform: scale(0);
  }

  .languageEnterActive {
    opacity: 1;
    transform: scale(1);
  }

  .languageEnter,
  .languageEnterActive,
  .languageExit {
    transition: opacity 300ms, transform 300ms;
  }

  .arrowEnter,
  .arrowExitDone {
    transform: rotate(0);
  }

  .arrowEnterActive,
  .arrowEnterDone,
  .arrowExit {
    transform: rotate(180deg);
  }

  .arrowExitActive {
    transform: rotate(360deg);
  }

  .arrowEnter,
  .arrowEnterActive,
  .arrowExit,
  .arrowExitActive {
    transition: transform 300ms;
  }

  .block {
    .button {
      @include flex(nowrap, row, flex-start, center);

      .wrapper {
        &:first-child {
          @include width-height(2rem, 2rem);
          margin-right: 1rem;
        }

        &:last-child {
          @include width-height(1.2rem, 1.2rem);
        }
      }
    }
  }

  .hidden {
    position: absolute;
    top: auto;
    bottom: calc(100% + 2rem);
    padding: 1rem;
    border-radius: 0.4rem;
    background-color: $link-background-color-active;

    &:before {
      content: "";
      position: absolute;
      bottom: -2rem;
      left: 50%;
      transform: translateX(-50%);
      border: 1rem solid transparent;
      border-top: 1rem solid $link-background-color-active;
    }

    .button {
      @include width-height(2rem, 2rem);

      &:not(:last-child) {
        margin-bottom: 1rem;
      }
    }
  }
}
