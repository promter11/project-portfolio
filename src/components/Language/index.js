import React from "react";
import { CSSTransition } from "react-transition-group";

import Arrow from "../../assets/icons/aside/arrow-down.svg";
import styles from "./Language.module.scss";

const Language = ({ state, actions }) => {
  const { toggle, defaultLanguage, languagesList } = state;
  const { setToggle, setLanguage } = actions;

  return (
    <div className={styles.language}>
      <div className={styles.block}>
        <button className={styles.button} onClick={() => setToggle()}>
          <div className={styles.wrapper}>
            <img
              className={styles.icon}
              src={defaultLanguage.icon}
              alt={defaultLanguage.language}
            />
          </div>
          <div className={styles.wrapper}>
            <CSSTransition
              in={toggle}
              timeout={300}
              classNames={{
                enter: styles.arrowEnter,
                enterActive: styles.arrowEnterActive,
                enterDone: styles.arrowEnterDone,
                exit: styles.arrowExit,
                exitActive: styles.arrowExitActive,
                exitDone: styles.arrowExitDone,
              }}
            >
              <img className={styles.arrow} src={Arrow} alt="Arrow" />
            </CSSTransition>
          </div>
        </button>
      </div>
      <CSSTransition
        in={toggle}
        timeout={300}
        classNames={{
          enter: styles.languageEnter,
          enterActive: styles.languageEnterActive,
          exit: styles.languageExit,
        }}
        unmountOnExit
      >
        <div className={styles.hidden}>
          {languagesList.map(({ language, icon }, index) => {
            return (
              <button
                key={index}
                className={styles.button}
                type="button"
                onClick={() => setLanguage(index)}
              >
                <img className={styles.icon} src={icon} alt={language} />
              </button>
            );
          })}
        </div>
      </CSSTransition>
    </div>
  );
};

export default Language;
