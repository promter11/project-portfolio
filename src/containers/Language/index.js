import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import * as LanguageActions from "../../store/actions/LanguageActions";

import Language from "../../components/Language";

const mapStateToProps = (state) => {
  return {
    state: state.LanguageReducer,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(LanguageActions, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Language);
