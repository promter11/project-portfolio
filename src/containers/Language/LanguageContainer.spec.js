import configureStore from "redux-mock-store";
import { shallow } from "enzyme";

import Language from "./index";

const configureMockStore = configureStore();

const initialState = {
  LanguageReducer: {
    toggle: false,
    defaultLanguage: {
      language: "RU",
      icon: "russia.svg",
    },
    languagesList: [
      {
        language: "RU",
        icon: "russia.svg",
      },
      {
        language: "EN",
        icon: "uk.svg",
      },
    ],
  },
};

const store = configureMockStore(initialState);

describe("Language container", () => {
  it("should renders correctly with props", () => {
    const wrapper = shallow(<Language store={store} />).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
