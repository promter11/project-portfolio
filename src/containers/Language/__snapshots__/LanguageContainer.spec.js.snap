// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language container should renders correctly with props 1`] = `
<Language
  actions={
    Object {
      "setLanguage": [Function],
      "setToggle": [Function],
    }
  }
  state={
    Object {
      "defaultLanguage": Object {
        "icon": "russia.svg",
        "language": "RU",
      },
      "languagesList": Array [
        Object {
          "icon": "russia.svg",
          "language": "RU",
        },
        Object {
          "icon": "uk.svg",
          "language": "EN",
        },
      ],
      "toggle": false,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
